protocol/x/clob/keeper/deleveraging.go

```diff
@@ -152,16 +165,24 @@ func (k Keeper) CanDeleverageSubaccount(
 		satypes.Update{SubaccountId: subaccountId},
 	)
 	if err != nil {
-		return false, err
+		return false, false, err
 	}
 
-	// Deleveraging cannot be performed if the subaccounts net collateral is non-negative.
-	if bigNetCollateral.Sign() >= 0 {
-		return false, nil
+	clobPairId, err := k.GetClobPairIdForPerpetual(ctx, perpetualId)
+	if err != nil {
+		return false, false, err
+	}
+	clobPair := k.mustGetClobPair(ctx, clobPairId)
+
+	// Negative TNC, deleverage at bankruptcy price.
+	if bigNetCollateral.Sign() == -1 {
+		return true, false, nil
 	}
```

Comment:
nit: move the fetching of the CLOB pair to after the `bigNetCollateral` sign check (since it's not used if you early return)?